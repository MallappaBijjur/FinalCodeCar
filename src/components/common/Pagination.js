import React from 'react';
import "./pagination.css";

const Pagination = props => {
	const { currentPageNo, getCarsPerPage, totalPageCount } = props;

	const handlePrevPage = () => {
		if (currentPageNo - 1 !== 0) {
	      getCarsPerPage(currentPageNo - 1);
	    } else {
	      getCarsPerPage(1)
	    }
	};

	const handleNextPage = () => {
		if (currentPageNo + 1 > totalPageCount) {
	      getCarsPerPage(totalPageCount);
	    } else {
	      getCarsPerPage(currentPageNo + 1);
	    }
	};

	return (<div className='pagination'>
		        <a
		          href='#available cars'
		          className='link__item--text link__item--text--small pagelink__text--orange'
		          onClick={() => getCarsPerPage(1)}
		        >First</a>
		        <a
		          href='#available cars'
		          className='link__item--text link__item--text--small pagelink__text--orange'
		          onClick={() => handlePrevPage()}
		        >Previous</a>
		        <a
		          href='#available cars'
		          className='link__item--text link__item--text--small link__text--black'
		        >{`Page ${currentPageNo} of ${totalPageCount}`}
		        </a>
		        <a
		          href='#available cars'
		          className='link__item--text link__item--text--small pagelink__text--orange'
		          onClick={() => handleNextPage()}
		        >Next</a>
		        <a
		          href='#available cars'
		          className='link__item--text link__item--text--small pagelink__text--orange'
		          onClick={() => getCarsPerPage(totalPageCount)}
		        >Last</a>
	      </div>)
};

export default Pagination;