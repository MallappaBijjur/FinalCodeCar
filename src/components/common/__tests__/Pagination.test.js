import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup, fireEvent } from '@testing-library/react';
import rerender from 'react-test-renderer';
import Pagination from '../Pagination';

describe("Pagination", () => {

	const originalError = console.error
	beforeAll(() => {
		console.error = (...args) => {
			if (/Warning.*not wrapped in act/.test(args[0])) {
				return
			}
			originalError.call(console, ...args)
		}
	})

	afterAll(() => {
		console.error = originalError
	});

	afterEach(cleanup);
	
	it("renders without crashing", ()=>{
		const div = document.createElement("div");
		ReactDOM.render(<Pagination />, div);
	});
	
	it('renders Pagination component', () => {
        const { getByText } = render(<Pagination />);   
        const linkElement = getByText(/First/i);
        expect(linkElement).toBeInTheDocument();
	});
	
	it("matches snapshot", ()=> {
		const tree = rerender.create(<Pagination />).toJSON();
		expect(tree).toMatchSnapshot();
    });
    
    it("should test handlePrevPage method middle page", ()=> {
        const currentPageNo = 5;
        const getCarsPerPage = jest.fn() ;
        const totalPageCount = 10;

        const { getByText } = render(<Pagination
                                        totalPageCount={totalPageCount}
                                        currentPageNo={currentPageNo}
                                        getCarsPerPage={getCarsPerPage}
                                    />);
        fireEvent.click(getByText("Previous"));
        expect(getCarsPerPage.mock.calls.length).toBe(1);
        expect(getCarsPerPage).toHaveBeenCalledWith(4);

    });

    it("should test handlePrevPage method when for the first page", ()=> {
        const currentPageNo = 1;
        const getCarsPerPage = jest.fn() ;
        const totalPageCount = 10;

        const { getByText } = render(<Pagination
                                        totalPageCount={totalPageCount}
                                        currentPageNo={currentPageNo}
                                        getCarsPerPage={getCarsPerPage}
                                    />);
        fireEvent.click(getByText("Previous"));
        expect(getCarsPerPage.mock.calls.length).toBe(1);
        expect(getCarsPerPage).toHaveBeenCalledWith(1);

    });

    it("should test handlePrevPage method when for the last page", ()=> {
        const currentPageNo = 10;
        const getCarsPerPage = jest.fn() ;
        const totalPageCount = 10;

        const { getByText } = render(<Pagination
                                        totalPageCount={totalPageCount}
                                        currentPageNo={currentPageNo}
                                        getCarsPerPage={getCarsPerPage}
                                    />);
        fireEvent.click(getByText("Previous"));
        expect(getCarsPerPage.mock.calls.length).toBe(1);
        expect(getCarsPerPage).toHaveBeenCalledWith(9);

    });

    it("should test handleNextPage method for first page", ()=> {
        const currentPageNo = 1;
        const getCarsPerPage = jest.fn() ;
        const totalPageCount = 10;

        const { getByText } = render(<Pagination
                                        totalPageCount={totalPageCount}
                                        currentPageNo={currentPageNo}
                                        getCarsPerPage={getCarsPerPage}
                                    />);
        fireEvent.click(getByText("Next"));
        expect(getCarsPerPage.mock.calls.length).toBe(1);
        expect(getCarsPerPage).toHaveBeenCalledWith(2);
    });

    it("should test handleNextPage method for middle page", ()=> {
        const currentPageNo = 5;
        const getCarsPerPage = jest.fn() ;
        const totalPageCount = 10;

        const { getByText } = render(<Pagination
                                        totalPageCount={totalPageCount}
                                        currentPageNo={currentPageNo}
                                        getCarsPerPage={getCarsPerPage}
                                    />);
        fireEvent.click(getByText("Next"));
        expect(getCarsPerPage.mock.calls.length).toBe(1);
        expect(getCarsPerPage).toHaveBeenCalledWith(6);
    });

    it("should test handleNextPage method for last page", ()=> {
        const currentPageNo = 10;
        const getCarsPerPage = jest.fn() ;
        const totalPageCount = 10;

        const { getByText } = render(<Pagination
                                        totalPageCount={totalPageCount}
                                        currentPageNo={currentPageNo}
                                        getCarsPerPage={getCarsPerPage}
                                    />);
        fireEvent.click(getByText("Next"));
        expect(getCarsPerPage.mock.calls.length).toBe(1);
        expect(getCarsPerPage).toHaveBeenCalledWith(10);
    });
});
