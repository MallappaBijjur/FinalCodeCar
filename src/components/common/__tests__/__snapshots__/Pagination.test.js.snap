// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination matches snapshot 1`] = `
<div
  className="pagination"
>
  <a
    className="link__item--text link__item--text--small pagelink__text--orange"
    href="#available cars"
    onClick={[Function]}
  >
    First
  </a>
  <a
    className="link__item--text link__item--text--small pagelink__text--orange"
    href="#available cars"
    onClick={[Function]}
  >
    Previous
  </a>
  <a
    className="link__item--text link__item--text--small link__text--black"
    href="#available cars"
  >
    Page undefined of undefined
  </a>
  <a
    className="link__item--text link__item--text--small pagelink__text--orange"
    href="#available cars"
    onClick={[Function]}
  >
    Next
  </a>
  <a
    className="link__item--text link__item--text--small pagelink__text--orange"
    href="#available cars"
    onClick={[Function]}
  >
    Last
  </a>
</div>
`;
